### PR TITLE
Misc market select fixes

### PR DESCRIPTION
**Pull request by bogberry**

### CODE REVIEW

**Comment by bogberry** [comment]
For something like the following

```typescript
export const useAllMarketPrices = (allMarketData: ApiMarket[]) => {
  return useQuery<{ market_id: number; price: number }[]>(
    ["allMarketPrices", allMarketData],
    async () => {
      const data: Promise<Orderbook>[] = [];
      allMarketData.forEach((market) => {
        data.push(
          fetch(
            new URL(`market/${market.market_id}/orderbook?depth=1`, API_URL)
              .href,
          ).then((res) => {
            return res.json();
          }),
        );
      });
      const orderBooks = await Promise.all(data);
      return orderBooks.map((orderBook, i) => {
        return {
          market_id: allMarketData[i].market_id,
          price: (orderBook.asks[0].price + orderBook.bids[0].price) / 2,
        };
      });
    },
  );
};
```

Instead of initializing an empty array of promises, and pushing promised onto the array in a for each loop, it's simpler just to use the map function.

You also seem to be mixing Promise-then syntax with async-await, so I think you'd be better off switching to async-await everywhere. The query key doesn't have to include the value of `allMarketData` either, so if you put all of that together you get something like this:

(Also, I changed the API endpoint to `/markets/:market_id`, sorry!)


```typescript
export const useAllMarketPrices = (allMarketData: ApiMarket[]) => {
  return useQuery<{ market_id: number; price: number }[]>(
    ["allMarketPrices"],
    async () => {
      const orderbooks: Orderbook[] = await Promise.all(
        allMarketData.map(async (market) => {
          const res = await fetch(
            new URL(`markets/${market.market_id}/orderbook?depth=1`, API_URL)
              .href,
          );
          const data = await res.json();
          return data as Orderbook;
        }),
      );
      return orderbooks.map((orderbook, i) => {
        return {
          market_id: allMarketData[i].market_id,
          price: (orderbook.asks[0].price + orderbook.bids[0].price) / 2,
        };
      });
    },
  );
};
```

Staying away from mutable objects and sticking to assigning a value once and never mutating it is an idea that comes from functional programming that can be helpful in preventing bugs.

On top of this, you could combine the two map functions and avoid having to retrieve `market_id` via index later, since you already have it in the original data you are mapping over.

```typescript
export const useAllMarketPrices = (allMarketData: ApiMarket[]) => {
  return useQuery<{ market_id: number; price: number }[]>(
    ["allMarketPrices"],
    async () => {
      return await Promise.all(
        allMarketData.map(async ({ market_id }) => {
          const res = await fetch(
            new URL(`markets/${market_id}/orderbook?depth=1`, API_URL).href,
          );
          const data: Orderbook = await res.json();
          return {
            market_id,
            price: (data.asks[0].price + data.bids[0].price) / 2,
          };
        }),
      );
    },
  );
};
```

But price isn't actually determined by taking the average between the highest bid and the lowest ask, so here you can just use the `close` price from the stats endpoint. Just thought this would be a good opportunity for some coding tips!

**Review by chiumax — APPROVED** [review]
other than what brian said, TY for the comment

**Comment by bogberry** [comment]
> Are these changes in the PR? I would agree that map is a bit more concise. That said, I think we should also be careful about Promise.all because if any of the calls throws and we don't handle it, we end up failing everything.

Yup good point, you should have error handling here, I just didn't go that far because I thought the code snippet above demonstrated my point & this code doesn't actually have to be in the codebase.

File: src/typescript/frontend/src/components/trade/DepositWithdrawModal/SelectMarketContent.tsx [hunk 1]
```diff
@@ -48,9 +47,7 @@ export const SelectMarketContent: React.FC<{
       columnHelper.accessor("market_id", {
         cell: (info) => (
           <PriceCell
-            price={
-              getPriceByMarketId(info.getValue(), marketPrices)?.price || 0
-            }
+            price={getStatsByMarketId(info.getValue(), marketStats)?.close || 0}
```
[the remark this answers is not in the document]
Comment (bogberry) [reply src/typescript/frontend/src/components/trade/DepositWithdrawModal/SelectMarketContent.tsx:50]: Yeah probably should display `"-"` or something, but I'll leave this for a different PR

**Pull request merged**